src/common/fault_domain.c

```diff
@@ -50,6 +66,58 @@ d_fd_tree_init(struct d_fd_tree *tree, const uint32_t *compressed,
 	return d_fd_tree_reset(tree);
 }
 
+static bool
+tree_has_fault_domain(struct d_fd_tree *tree)
+{
+	return (tree->fdt_compressed[0] & D_FD_TREE_HAS_FAULT_DOMAIN) != 0;
```

Comment:
just confirm that the D_FD_TREE_HAS_FAULT_DOMAIN bit only set when the DAOS_PROP_PO_REDUN_LVL property is not on NODE and also not on RANK?
i.e. if DAOS_PROP_PO_REDUN_LVL on NODE or RANK, should not carry D_FD_TREE_HAS_FAULT_DOMAIN bit in the first byte?

Reply:
Correct, if NODE is selected as the pool redundancy level, the bit will be unset. Only if a higher level is chosen during pool create will this bit be set.